### PR TITLE
Don't choose banks with no valid mandates for lsv/dd orders.

### PR DESCRIPTION
**Pull request by ecino**
Override line2bank to avoid choosing a bank that don't have valid mandates.

### CODE REVIEW

**Comment by nbessi** [comment]
@ecino @davidZIVI can you sign the CLA please

File: l10n_ch_lsv_dd/model/move_line.py [hunk 1]
```diff
@@ -39,3 +39,29 @@ def _is_generate_bvr(self, cr, uid, invoice, context=None):
         return val or (invoice.partner_bank_id and
                        invoice.partner_bank_id.state == 'iban' and
                        invoice.partner_bank_id.lsv_identifier)
+
+    def line2bank(self, cr, uid, ids, payment_mode_id, context=None):
+        ''' Override line2bank to avoid choosing a bank that has only
+            cancelled mandate.
+        '''
+        pay_mode_obj = self.pool.get('payment.mode')
+        if payment_mode_id:
```
Comment (nbessi) [inline l10n_ch_lsv_dd/model/move_line.py:48]: 9 levels of loop condition imbrication is not really maintainable. 
Can you please breack this code in smaller chunck.

Comment (nbessi) [inline l10n_ch_lsv_dd/model/move_line.py:48]: can you please also give me a pointer to the definition of `suitable_bank_types`

**Comment by ecino** [comment]
We signed the CLA and sent them.

**Comment by gurneyalex** [comment]
@ecino I updated the OCA database with you github login

However, I did not find a CLA for @davidZIVI but maybe I'm not looking under the correct name. There is noone called 'Zivi' in the database...

**Comment by ecino** [comment]
We just sent the contract by e-mail this morning.

**Comment by ecino** [comment]
We made the corrections, and davidZIVI also signed the CLA, can you acknowledge that ?

File: l10n_ch_lsv_dd/model/move_line.py [hunk 2]
```diff
@@ -39,3 +39,42 @@ def _is_generate_bvr(self, cr, uid, invoice, context=None):
...
+                            cr, uid,
+                            line.partner_id.bank_ids,
+                            bank_types, context)
+                        if bank_id:
+                            line2bank[line.id] = bank_id
+                        else:
+                            line2bank.update(
+                                super(account_move_line, self).line2bank(
+                                    cr, uid, [line.id],
+                                    payment_mode_id, context=None))
```
Comment (yvaucher) [inline l10n_ch_lsv_dd/model/move_line.py]: You certainly mean `context=context`

File: l10n_ch_lsv_dd/model/move_line.py [hunk 3]
```diff
@@ -39,3 +39,42 @@ def _is_generate_bvr(self, cr, uid, invoice, context=None):
...
+                return line2bank
+        return super(
+            account_move_line, self).line2bank(
+                cr, uid, ids, payment_mode_id, context=None)
```
Comment (yvaucher) [inline l10n_ch_lsv_dd/model/move_line.py]: Same here please forward the context

**Comment by yvaucher** [comment]
@ecino 

Just 2 little remarks,

Thanks for having signed the CLA

**Comment by ecino** [comment]
Thanks for the review, I corrected this.

**Comment by yvaucher** [comment]
:+1:

**Comment by nbessi** [comment]
:+1:

**Pull request merged**